### PR TITLE
fix: align setup of resource and data source id for testing framework

### LINKED ISSUE #292: Unify spec and initialization of deprecated id attributes
The plugin testing framework requires every resource/datasource to have an "id" attribute. Therefore we add a dummy "id" attribute if no sensible id is provided by the BTP. However, currently the definitions of these "id" attributes are very diverse. We need to unify their specification and initialization.

In the schema, every "id" attribute that is only required by the testing framework needs to:

- be specified as "Computed"
- be deprecated by providing a proper "DeprecationMessage" (maybe we can use the same message everywhere)
- have a meaningful "MarkdownDescription"
- use stringplanmodifier.UseStateForUnknown() if update should be supported

In the Create operation, the "id" attribute needs to be filled with an unique value, e.g. for directory role-collections we use "directory id,role-collection name".

### PR DESCRIPTION
**Pull request by lechnerc77**
## Purpose

This PR contains the changes to have an aligned setup of the **artificial** `ID` field in resources and data sources that is needed for the Terraform plugin test framework. The changes comprise:

* Mark the ID field as "computed"
* Align the "DeprecationMessage" field
* Align the "MarkdownDescription"
* Use `stringplanmodifier.UseStateForUnknown()` in schema
* Regenerate documentation
* re-record test fixtures for resources

Closes #292 

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[X] Bugfix
[ ] Feature
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
go test ./...
```

## What to Check

Verify that the following are valid

* All tests are executed successfully

## Other Information

Further information is available here: <https://github.com/hashicorp/terraform-plugin-testing/issues/84>

### CODE REVIEW

File: internal/provider/datasource_directory_entitlements.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ To view all the resources that a directory and its subdirectories and subaccount
 					uuidvalidator.ValidUUID(),
 				},
 			},
-			"id": schema.StringAttribute{
+			"id": schema.StringAttribute{ // required hashicorps terraform plugin testing framework
```
Comment (kuntzed) [inline internal/provider/datasource_directory_entitlements.go]: required by?

Comment (lechnerc77) [reply internal/provider/datasource_directory_entitlements.go]: Fixed

File: internal/provider/resource_directory_role.go [hunk 2]
```diff
@@ -49,10 +51,13 @@ __Further documentation:__
 					uuidvalidator.ValidUUID(),
 				},
 			},
-			"id": schema.StringAttribute{ // required hashicorps terraform plugin testing framework
-				DeprecationMessage:  "Use the `directory_id` attribute instead",
-				MarkdownDescription: "The ID of the directory.",
+			"id": schema.StringAttribute{ // required by hashicorps terraform plugin testing framework
+				DeprecationMessage:  "Use the `directory_id`, `name`, `app_id` and `role_template_name` attributes instead",
```
Comment (kuntzed) [inline internal/provider/resource_directory_role.go]: The order as concatenated in the "id" is `directory_id`, `name`,  `role_template_name` and `app_id`. Don't know if this should be the same.

Comment (lechnerc77) [reply internal/provider/resource_directory_role.go]: Good catch, fixed.

Comment (kuntzed) [reply internal/provider/resource_directory_role.go]: probably you want to do the same for subaccount? sorry for only commenting for directory.

Comment (lechnerc77) [reply internal/provider/resource_directory_role.go]: Now we have it fixed

**Review by kuntzed — APPROVED** [review]

**Pull request merged**